eppoclient/lruassignmentlogger_test.go

```diff
@@ -0,0 +1,112 @@
+		Variation:         "123.45",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	}
+
+	logger.LogAssignment(event)
+	logger.LogAssignment(event)
+
+	innerLogger.AssertNumberOfCalls(t, "LogAssignment", 1)
+}
+
+func Test_LruAssignmentLogger_panicsAreNotCached(t *testing.T) {
+	innerLogger := new(mockLogger)
+	innerLogger.On("LogAssignment", mock.Anything).Panic("test panic")
+
+	logger := NewLruAssignmentLogger(innerLogger, 1000)
+
+	event := AssignmentEvent{
+		Experiment:        "testExperiment",
+		FeatureFlag:       "testFeatureFlag",
+		Allocation:        "testAllocation",
+		Variation:         "123.45",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	}
+
+	assert.Panics(t, func() {
+		logger.LogAssignment(event)
+	})
+	assert.Panics(t, func() {
+		logger.LogAssignment(event)
+	})
+
+	innerLogger.AssertNumberOfCalls(t, "LogAssignment", 2)
+}
+
+func Test_LruAssignmentLogger_changeInVariationCausesLogging(t *testing.T) {
+	innerLogger := new(mockLogger)
+	innerLogger.On("LogAssignment", mock.Anything).Return()
+
+	logger := NewLruAssignmentLogger(innerLogger, 1000)
+
+	logger.LogAssignment(AssignmentEvent{
+		Experiment:        "testExperiment",
+		FeatureFlag:       "testFeatureFlag",
+		Allocation:        "testAllocation",
+		Variation:         "variation1",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	})
+	logger.LogAssignment(AssignmentEvent{
+		Experiment:        "testExperiment",
+		FeatureFlag:       "testFeatureFlag",
+		Allocation:        "testAllocation",
+		Variation:         "variation2",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	})
+
+	innerLogger.AssertNumberOfCalls(t, "LogAssignment", 2)
+}
+
+func Test_LruAssignmentLogger_changeInAllocationCausesLogging(t *testing.T) {
+	innerLogger := new(mockLogger)
+	innerLogger.On("LogAssignment", mock.Anything).Return()
+
+	logger := NewLruAssignmentLogger(innerLogger, 1000)
+
+	logger.LogAssignment(AssignmentEvent{
+		Experiment:        "testExperiment",
+		FeatureFlag:       "testFeatureFlag",
+		Allocation:        "testAllocation1",
+		Variation:         "variation",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	})
+	logger.LogAssignment(AssignmentEvent{
+		Experiment:        "testExperiment",
+		FeatureFlag:       "testFeatureFlag",
+		Allocation:        "testAllocation2",
+		Variation:         "variation",
+		Subject:           "testSubject",
+		Timestamp:         "testTimestamp",
+		SubjectAttributes: SubjectAttributes{"testKey": "testValue"},
+	})
+
+	innerLogger.AssertNumberOfCalls(t, "LogAssignment", 2)
```

Comment:
suggestion: one more invocation, after a change, to verify that only 2 calls

Reply:
Added this as another test case + added more test cases to ensure that each `[allocation,variation]` pair is only logged once